nodebuilder/p2p/addrs.go

```diff
@@ -6,18 +6,29 @@ import (
 	p2pconfig "github.com/libp2p/go-libp2p/config"
 	hst "github.com/libp2p/go-libp2p/core/host"
 	ma "github.com/multiformats/go-multiaddr"
+	"golang.org/x/exp/slices"
 )
 
 // Listen returns invoke function that starts listening for inbound connections with libp2p.Host.
 func Listen(listen []string) func(h hst.Host) (err error) {
 	return func(h hst.Host) (err error) {
-		maListen := make([]ma.Multiaddr, len(listen))
-		for i, addr := range listen {
-			maListen[i], err = ma.NewMultiaddr(addr)
+		maListen := make([]ma.Multiaddr, 0, len(listen))
+		for _, addr := range listen {
+			maddr, err := ma.NewMultiaddr(addr)
 			if err != nil {
 				return fmt.Errorf("failure to parse config.P2P.ListenAddresses: %s", err)
 			}
+
+			// TODO(@WonderTan): Remove this check when QUIC is stable
+			if slices.ContainsFunc(maddr.Protocols(), func(p ma.Protocol) bool {
+				return p.Code == ma.P_QUIC_V1 || p.Code == ma.P_WEBTRANSPORT
+			}) {
+				continue
+			}
```

Comment:
This is to prevent the following logs, yet keeping addresses in config:
```
2023-10-11T19:12:01.310+0200	WARN	swarm2	swarm/swarm_listen.go:30	listening failed	{"on": "/ip4/0.0.0.0/udp/2121/quic-v1/webtransport", "error": "no transport for protocol"}
2023-10-11T19:12:01.310+0200	WARN	swarm2	swarm/swarm_listen.go:30	listening failed	{"on": "/ip6/::/udp/2121/quic-v1/webtransport", "error": "no transport for protocol"}
2023-10-11T19:12:01.310+0200	WARN	swarm2	swarm/swarm_listen.go:30	listening failed	{"on": "/ip4/0.0.0.0/udp/2121/quic-v1", "error": "no transport for protocol"}
2023-10-11T19:12:01.311+0200	WARN	swarm2	swarm/swarm_listen.go:30	listening failed	{"on": "/ip6/::/udp/2121/quic-v1", "error": "no transport for protocol"}
```